kernel/src/engine/arrow_expression/evaluate_expression.rs

```diff
@@ -120,28 +124,15 @@ pub fn evaluate_expression(
         (Predicate(_), Some(data_type)) => Err(Error::generic(format!(
             "Predicate evaluation produces boolean output, but caller expects {data_type:?}"
         ))),
-        (Unary(UnaryExpression { op, expr }), data_type) => {
-            match op {
-                ToJson => {
-                    if let Some(data_type) = data_type {
-                        if data_type != &DataType::STRING {
-                            return Err(Error::generic(format!(
-                                "ToJson operator requires STRING output, but got {data_type:?}"
-                            )));
-                        }
-                    }
-                }
-            };
-
-            let input_arr = evaluate_expression(expr, batch, None)?;
-
-            type Operation = fn(&dyn Datum) -> Result<ArrayRef, ArrowError>;
-            let eval: Operation = match op {
-                ToJson => to_json,
-            };
-
-            Ok(eval(&input_arr)?)
-        }
+        (Unary(UnaryExpression { op: ToJson, expr }), result_type) => match result_type {
+            None | Some(&DataType::STRING) => {
+                let input = evaluate_expression(expr, batch, None)?;
+                to_json(&input).map_err(Into::into)
```

Comment:
Isn't this the same as
```suggestion
                Ok(to_json(&input)?)
```
(I'm pretty sure `?` uses `Into` under the hood)